hal/src/rtl872x/deviceid_hal.cpp

```diff
@@ -56,9 +57,20 @@ int readLogicalEfuse(uint32_t offset, uint8_t* buf, size_t size) {
 
     memset(efuseBuf, 0xff, LOGICAL_EFUSE_SIZE);
 
-    auto ret = EFUSE_LMAP_READ(efuseBuf);
-    CHECK_TRUE(ret == EFUSE_SUCCESS, SYSTEM_ERROR_INTERNAL);
+    bool dataConsistent = false;
+    for (int i = 0; i < 5 && !dataConsistent; i++) {
+        EFUSE_LMAP_READ(efuseBuf);
+        uint32_t crc1 = HAL_Core_Compute_CRC32(efuseBuf, LOGICAL_EFUSE_SIZE);
```

Comment:
How do we want to handle this call for bootloader?
1. Add `HAL_Core_Compute_CRC32` to bootloader / expose `Compute_CRC32`?
2. Exclude this check from bootloader and accept the 4.5% failure rate?
3. include another implementation of `Compute_CRC32` in bootloader like we do for prebootloader

Reply:
1 is the right choice.